concourse/pipelines/create-cloudfoundry.yml

```diff
@@ -5524,6 +5535,209 @@ jobs:
+        trigger: true
+
+      - in_parallel:
+        - <<: *get-paas-cf
+          trigger: true
+          passed: ['post-deploy']
+
+      # this tests the "integration" of this paas-billing revision with the
+      # configuration in this paas-cf revision, performed before deployment
+      # to give us a chance to avoid deploying a broken combination.
+      - task: paas-billing-experimental-integration-tests
+        tags: [colocated-with-web]
+        config:
+          platform: linux
+          image_resource: *cf-acceptance-tests-image-resource
+          params:
+            AWS_REGION: ((aws_region))
+            DEPLOY_ENV: ((deploy_env))
+          inputs:
+            - name: paas-cf
+            - name: paas-billing-experimental
+              path: src/github.com/alphagov/paas-billing
+          run:
+            path: sh
+            args:
+              - -e
+              - -u
+              - -c
+              - |
+                mkdir ./src/github.com/alphagov/paas-billing/gherkin/features/
+
+                cp "paas-cf/config/billing/output/${AWS_REGION}.json" ./src/github.com/alphagov/paas-billing/config.json
+                cp "paas-cf/config/billing/tests/${AWS_REGION}_billing_rds_charges.feature" ./src/github.com/alphagov/paas-billing/gherkin/features/
+
+                wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
+                echo "deb http://apt.postgresql.org/pub/repos/apt/ bionic-pgdg main" | tee /etc/apt/sources.list.d/pgdg.list
+
+                apt-get -q update
+                apt-get -q -y install postgresql-12 postgresql-client-12
+
+                cat > /etc/postgresql/12/main/pg_hba.conf <<-EOF
+                local   all   postgres   trust
+                host    all   postgres   127.0.0.1/32   trust
+                EOF
+
+                pg_ctlcluster 12 main start
+
+                export TEST_DATABASE_URL="postgres://postgres:@localhost:5432/?sslmode=disable"
+                cd src/github.com/alphagov/paas-billing
+                export APP_ROOT="$PWD"
+
+                make integration
+
+      - task: deploy-paas-billing-experimental
+        tags: [colocated-with-web]
+        config:
+          platform: linux
+          image_resource: *cf-cli-image-resource
+          params:
+            AWS_REGION: ((aws_region))
+            SYSTEM_DNS_ZONE_NAME: ((system_dns_zone_name))
+            DEPLOY_ENV: ((deploy_env))
+            CF_CLIENT_SECRET: ((cf_client_secret))
+            LOGIT_ADDRESS: ((logit_syslog_address))
+            LOGIT_PORT: ((logit_syslog_port))
+            API_ENDPOINT: ((api_endpoint))
+            CF_ADMIN: ((cf_admin))
+            CF_PASS: ((cf_pass))
+          inputs:
+            - name: paas-cf
+            - name: paas-billing-experimental
+              path: paas-billing
+          run:
+            path: sh
+            args:
+              - -e
+              - -u
+              - -c
+              - |
```

Comment:
Just thinking about cost.... is there any easy way to just put this experimental version in just a limited number of environments?

Reply:
I considered this, but again,
 - new-billing didn't do it
 - it provisions smaller databases for non-prod (i could probably reduce this further actually)
 - this would introduce complexity and an extra piece of configuration that I'd rather avoid.

Reply:
Okay. Thought I'd check.